groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -117,6 +116,7 @@ BSLS_IDENT("$Id: $")
 #include <bsls_util.h>
 
 #include <stddef.h>
+#include "bslstl_inplace.h"
 
 #ifdef BSLS_LIBRARYFEATURES_HAS_CPP11_BASELINE_LIBRARY
 //#include <bsl_type_traits.h>
```

Comment:
Is there clarity on handling this?  IIRC, Mike suggested that using platform `type_traits` is basically fine in this case?

Reply:
Yes, in this case ti is ok. 
- I can't include bsl_type_traits.h in a bslstl header. 
- None of the traits from type_traits are used in pre C++11 code. 
- All the traits that are available in bslmf are picked up from bslmf. 

I've removed the commented out include